smurfs/src/reducers/index.js

```diff
@@ -1,23 +1,26 @@
-/*
-  Be sure to import in all of the action types from `../actions`
-*/
+import { LOADING, SUCCESS, ERROR } from '../actions/index';
 
-/*
- Your initial/default state for this project could *Although does not have to* look a lot like this
- {
+
+ const initialState = {
    smurfs: [],
-   fetchingSmurfs: false
-   addingSmurf: false
-   updatingSmurf: false
-   deletingSmurf: false
+   fetchingSmurfs: false,
+   addingSmurf: false,
+   updatingSmurf: false,
+   deletingSmurf: false,
+   fetchingSmurfs: false,
    error: null
  }
-*/
 
-/*
-  You'll only need one smurf reducer for this project.
-  Feel free to export it as a default and import as rootReducer. 
-  This will guard your namespacing issues.
-  There is no need for 'combineReducers' in this project.
-  Components can then read your store as, `state` and not `state.fooReducer`.
-*/
+
+export default (state=initialState, action) => {
+  switch(action.type){
+    case LOADING:
```

Comment:
In the case of the Loading and Success I'd have the error property set to null as well.